Sources_v3/StreamChatUI/ChatChannel/ChatMessageThreadInfoView.swift

```diff
@@ -0,0 +1,132 @@
+        shape.fillColor = nil
+        shape.lineWidth = 1.0 / UIScreen.main.scale
+    }
+
+    public var isLeftToRight: Bool {
+        let isLeftToRightWithTrailing = direction == .toTrailing && traitCollection.layoutDirection == .leftToRight
+        let isRightToLeftWithLeading = direction == .toLeading && traitCollection.layoutDirection == .rightToLeft
+        return isLeftToRightWithTrailing || isRightToLeftWithLeading
+    }
+
+    override open func draw(_ rect: CGRect) {
+        let corner: CGFloat = 16
+        let height = bounds.height
+
+        let startX = isLeftToRight ? 0 : bounds.width
+        let endX = isLeftToRight ? corner : (bounds.width - corner)
+
+        let path = CGMutablePath()
+        path.move(to: CGPoint(x: startX, y: 0))
+        path.addLine(to: CGPoint(x: startX, y: height - corner))
+        path.addQuadCurve(
+            to: CGPoint(x: endX, y: height),
+            control: CGPoint(x: startX, y: height)
+        )
+        shape.path = path
+        super.draw(rect)
+    }
+}
+
+open class ChatMessageThreadInfoView<ExtraData: UIExtraDataTypes>: View {
+    public var message: _ChatMessageGroupPart<ExtraData>? {
+        didSet { updateContentIfNeeded() }
+    }
+
+    public var onTap: (_ChatMessageGroupPart<ExtraData>?) -> Void = { _ in }
+
+    public private(set) lazy var avatarView = AvatarView().withoutAutoresizingMaskConstraints
+    public private(set) lazy var replyCountLabel: UILabel = {
+        let label = UILabel().withoutAutoresizingMaskConstraints
+        label.font = .preferredFont(forTextStyle: .footnote)
+        label.adjustsFontForContentSizeCategory = true
+        label.text = L10n.Message.Threads.reply
+        label.textColor = .messageThreadReply
+        return label
+    }()
+
+    public private(set) lazy var stack: UIStackView = {
+        let stack = UIStackView(arrangedSubviews: [avatarView, replyCountLabel]).withoutAutoresizingMaskConstraints
+        stack.distribution = .fill
+        stack.alignment = .center
+        stack.axis = .horizontal
+        stack.spacing = UIStackView.spacingUseSystem
+        return stack
+    }()
+
+    override open func setUp() {
+        super.setUp()
+        addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(didTap)))
+    }
+
+    override open func setUpLayout() {
+        super.setUpLayout()
+        embed(stack)
+        avatarView.widthAnchor.constraint(equalToConstant: 16).isActive = true
+        avatarView.heightAnchor.constraint(equalToConstant: 16).withPriority(999).isActive = true
+        replyCountLabel.setContentCompressionResistancePriority(.required, for: .horizontal)
+        replyCountLabel.setContentCompressionResistancePriority(.required, for: .vertical)
+    }
+
+    override open func updateContent() {
+        super.updateContent()
+        guard message?.parentMessageId == nil else {
+            updateForThreadReply()
+            return
+        }
+        updateForThreadStart()
```

Comment:
nit: what's wrong with using `if else` here?

Reply:
Originally it was single method, where I first went with removing unwanted branch via `guard` to stick with happy path.
Later it was refactored in multiple methods but guard stay.